### DISCUSSION TITLE
Inconsistencies regarding the sorting of structure elements and metadata

### DISCUSSION THREAD

**Discussion by BartChris** [kitodo/kitodo-production#4957] [Ideas]
We stumbled over the fact that for structural elements the decision was made, that their order can be defined via the ruleset or in alphabetical order. (https://github.com/kitodo/kitodo-production/issues/3413#issuecomment-756625021 , https://github.com/kitodo/kitodo-production/pull/4190 ) 
In the case of metadata a different decision was made insofar as the metadata elements are always sorted alphabetically. ( https://github.com/kitodo/kitodo-production/pull/4664 ). 

This behaviour appears inconsistent. There are scenarios were the ruleset allows the addition of many metadata elements but only some of the elements are used in most cases. Other elements are used only in special cases. Right now there is no way to define in the ruleset that the most used metadata elements should appear on top since the ruleset ordering is not considered in the UI. It would be preferable if the setting for structural elements does also apply for metadata.

@matthias-ronge @solth @andre-hohmann - could you maybe reconsider this behaviour?

**Comment by andre-hohmann** [comment]
In general, @BartChris is right: the behaviour is inconsistent and the order opportunities of structural elements and metadata should be the same. 
In particular, we only apply the alphabetically order of metadata and i did not notice this inconsistency. Therefore, an adjustment is for us currently not of high priority. 

I can only comment from the users perspective, but i cannot assess, if the behaviour can be adjusted.

**Comment by matthias-ronge** [comment]
The idea was: The order should follow the order of the `<permit>`-rules of the rule set. The point is, that the metadata should typically be recorded in an order that people can understand, for example: author, year, title, series, publisher, place of publication (as you know it from bibliographic citations).

If the `<restriction>`-rule is set to `unspecified="unrestricted"` (which is the default), the elements for which `<permit>`-rules exist, are given in `<permit>`-rule order, the remaining elements below in alphabetical order. This makes sense especially for option values, so that one can "pull up" from a long list of values ​​(e.g. all languages) the most common ones for the inventory to be digitized, and the rest will then appear below.

**Example:**

_List of languages:_

- Afar
- Abkhazian
- Achinese
- …
- Zaza

_Rule:_

```xml
<permit key="language" unspecified="unrestricted">
    <permit value="ger"/>
    <permit value="eng"/>
    <permit value="lat"/>
</permit>
```

_Sort order of options showing:_

- German
- English
- Latin
- Afar
- Abkhazian
- Achinese
- …

You can override this ruleset-internal sorting by specifying the "alphabetical" sorting property for the selected rule set in the rule set configuration of the project settings. In this case, everything is sorted alphabetically. Any behavior that differs from these two cases should be considered a bug.

**Comment by BartChris** [comment]
I would then consider this a bug and created an issue https://github.com/kitodo/kitodo-production/issues/4967
